### PR TITLE
7201 web form skip

### PR DESCRIPTION
**Pull request by DevneyHamilton**

### CODE REVIEW

File: spec/features/response_skip_rule_spec.rb [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+
+    form.questionings.each do |qing|
+      # Get instance count for parent instance.
+      inst_count = if qing.depth == 1
+        1
+      else
+        # TODO: When we add support for nested groups to this spec, we will need to refine this selector
+        # to distinguish between different sets of subinstances within a parent repeat group's instances.
+        page.all(%Q{div.qing-group-instance[data-group-id="#{qing.parent_id}"]}).size
+      end
```
[the remark this answers is not in the document]
Comment (DevneyHamilton) [reply spec/features/response_skip_rule_spec.rb]: So . . this is an if/else clause that's assigned a variable. Not sure what is preferred here.

File: spec/features/response_skip_rule_spec.rb [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+        list[qing] ||= []
+        list[qing] << inst
+      end
+    end
+
+    form.questionings.each do |qing|
+      # Get instance count for parent instance.
+      inst_count = if qing.depth == 1
+        1
+      else
```
[the remark this answers is not in the document]
Comment (DevneyHamilton) [reply spec/features/response_skip_rule_spec.rb]: Same as "So . . this is an if/else clause that's assigned a variable. Not sure what is preferred here."

File: spec/features/response_skip_rule_spec.rb [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+        end
+        list[qing] ||= []
+        list[qing] << inst
+      end
+    end
+
+    form.questionings.each do |qing|
+      # Get instance count for parent instance.
+      inst_count = if qing.depth == 1
+        1
```
[the remark this answers is not in the document]
Comment (DevneyHamilton) [reply spec/features/response_skip_rule_spec.rb]: Similar issue - the 11 spaces is referring to a {}.tap that is assigned to a variable. Not sure what is preferred here.

File: app/serializers/response_condition/form_item_serializer.rb [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+    def initialize(object, response_condition_group:)
+      super(object)
+      self.response_condition_group = response_condition_group
+    end
+
+    def condition_group
+      ResponseCondition::ConditionGroupSerializer.new(response_condition_group)
+    end
+  end
```
Comment (hooverlunch) [inline app/serializers/response_condition/form_item_serializer.rb]: Very very nice.

File: spec/support/contexts/response_form_conditional_logic.rb [hunk 5]
```diff
@@ -0,0 +1,108 @@
+# frozen_string_literal: true
+
+shared_context "response form conditional logic" do
```
Comment (hooverlunch) [inline spec/support/contexts/response_form_conditional_logic.rb]: Nice job moving this into a shared file. You don't actually need the module part if you're doing `shared_context`. Only if you have a file of helpers methods that you want included in _all_ specs of a certain type.

I recently reorg'd the spec/support folder to make this clearer...

File: app/assets/javascripts/views/response_condition_group_checker.js.coffee [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+    @results().indexOf(true) != -1
+
+  applyNegation: (bool) ->
+    if @conditionGroup.negate
+      !bool
+    else
+      bool
+
+  results: ->
+    @checkers.map (c) -> c.evaluate()
```
Comment (hooverlunch) [inline app/assets/javascripts/views/response_condition_group_checker.js.coffee]: Sweet work here!

File: spec/features/response_skip_logic_spec.rb [hunk 7]
```diff
@@ -0,0 +1,188 @@
...
+      fill_and_expect_visible(:text2, "B", visible - %i[text3 text4 rptq1])
+      fill_and_expect_visible(:text2, "Undo Skip", visible - %i[rptq1])
+      fill_and_expect_visible(:text3, "ShowRepeat", visible)
+      fill_and_expect_visible(:rptq1, "Fill Repeat", visible)
+      fill_and_expect_visible(:text2, "B", visible - %i[text3 text4 rptq1])
+      fill_and_expect_visible(:text2, "Undo Skip", visible)
+      fill_and_expect_visible(:text3, "HideRepeat", visible - %i[rptq1])
+    end
+  end
+end
```
Comment (hooverlunch) [inline spec/features/response_skip_logic_spec.rb]: Awesome spec!!

**Pull request merged**